uui-core/src/data/processing/views/treeHelpers.ts

```diff
@@ -0,0 +1,73 @@
+import { IMap } from "../../../types";
+import { CompositeKeysMap } from "./CompositeKeysMap";
+
+export interface TreeParams<TItem, TId> {
+    getId?(item: TItem): TId;
+    getParentId?(item: TItem): TId | undefined;
+    complexIds?: boolean;
+}
+
+export function newMap<TKey, TValue>(params: TreeParams<any, any>) {
+    if (params.complexIds) {
+        return new CompositeKeysMap<TKey, TValue>();
+    } else {
+        return new Map<TKey, TValue>();
+    }
+}
+
+export const deleteFromList = <TId>(id: TId, list: TId[]) => {
+    const foundIndex = list.findIndex((childId) => childId === id);
+    if (foundIndex !== -1) {
+        list.splice(foundIndex, 1);
+    }
+
+    return list;
+};
+
+const pasteNewItemIntoList = <TItem, TId>(
+    id: TId,
+    newItem: TItem,
+    byId: IMap<TId, TItem>,
```

Comment:
Maybe just pass the Tree as arguments here and in other helpers, instead of passing maps? This would simplify parameters, and don't break encapsulation (as these maps are implementation detail)